src/client/pscp_sftp.c

```diff
@@ -98,6 +98,13 @@ struct pscp_sftp_session*  startSftpSession(const char * sftp_url, const char *
     if(status == CURLE_OK) status = curl_easy_setopt(sftp->curl, CURLOPT_SSH_PRIVATE_KEYFILE, privateKeyPath);
     if(status == CURLE_OK)
     {
+        // Workaround for RedHat bug 1260742 - curl requiring public key
+        char pubKeyPath[PATH_MAX];
```

Comment:
if the string is  PATH_MAX + 1 in length, but only copying at most PATH_MAX, will that guarantee a NULL terminatior?  Or does pubKeyPath need to be PATH_MAX+1 legnth to always guarantee a NULL terminator?

memset the pubKeyPath to all 0x0 to start with?

Reply:
snprintf guarantees a NULL terminator if the input string is larger than the bound.

From the man page:
The functions snprintf() and vsnprintf() write at most size bytes (including the trailing null byte ('\0')) to str.

I verified this with some code:

```
#include <stdio.h>
int main()
{
    char* largeString = "abcdefghijklmnop";
    char smallstr[5];
    snprintf(smallstr, 5, "%s", largeString);

    printf("STR: %s\n", largeString);
    for(int i = 0; i < 5; i++)
    {
        printf("%d: %d\n", i, smallstr[i]);
    }
    return 0;
}
```
```
> ./a.out
STR: abcdefghijklmnop
0: 97
1: 98
2: 99
3: 100
4: 0

```

Reply:
nice, looks like my interpretation of the man page did not match the implementation; I still don't think it is exactly that clear, but the code sample says all that needs saying.